drivers/ieee802154/ieee802154_efr32.c

```diff
@@ -0,0 +1,558 @@
+/* ieee802154_efr32.c - Silabs EFR32 driver */
+
+/*
+ * Copyright (c) 2018 Evry ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME ieee802154_efr32
+#define LOG_LEVEL CONFIG_IEEE802154_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <irq.h>
+#include <net/ieee802154_radio.h>
+#include <net/net_if.h>
+#include <net/net_pkt.h>
+#include <misc/byteorder.h>
+#include <random/rand32.h>
+
+#include "rail_types.h"
+#include "em_core.h"
+#include "em_system.h"
+
+#include "rail.h"
+#include "rail_ieee802154.h"
+#include "pa_conversions_efr32.h"
+
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+#include <net/openthread.h>
+#endif
+
+#define RADIO_CONFIG_XTAL_FREQUENCY 38400000UL
+
+RAIL_DECLARE_TX_POWER_VBAT_CURVES(piecewiseSegments, curvesSg, curves24Hp, curves24Lp);
+
+#define ACK_TIMEOUT K_MSEC(10)
+
+enum EfrState
+{
+    EFR32_STATE_RX,
+    EFR32_STATE_TX,
+    EFR32_STATE_DISABLED,
+    EFR32_STATE_CCA,
+    EFR32_STATE_SLEEP,
+};
+
+static enum EfrState efr_state;
+
+enum
+{
+    IEEE802154_MAX_LENGTH = 127,
+    EFR32_FCS_LENGTH = 2,
+};
+
+struct efr32_context
+{
+    struct net_if *iface;
+    u8_t mac_addr[8];
+
+    u8_t rx_buf[IEEE802154_MAX_LENGTH];
+    u8_t tx_buf[IEEE802154_MAX_LENGTH];
+
+    K_THREAD_STACK_MEMBER(rx_stack,
+                          CONFIG_IEEE802154_EFR32_RX_STACK_SIZE);
+    struct k_thread rx_thread;
+
+    /* CCA complete sempahore. Unlocked when CCA is complete. */
+    struct k_sem cca_wait;
+    /* RX synchronization semaphore. Unlocked when frame has been
+	 * received.
+	 */
+    struct k_sem rx_wait;
+    /* TX synchronization semaphore. Unlocked when frame has been
+	 * sent or CCA failed.
+	 */
+    struct k_sem tx_wait;
+
+    /* TX result. Set to 1 on success, 0 otherwise. */
+    bool tx_success;
+
+    u8_t channel;
+
+    RAIL_Handle_t rail_handle;
+};
+
+static struct efr32_context efr32_data;
+
+const RAIL_IEEE802154_Config_t rail_ieee802154_config = {
+    .addresses = NULL,
+    .ackConfig = {
+        .enable = true,
+        .ackTimeout = 894,
+        .rxTransitions = {
+            RAIL_RF_STATE_RX,
+            RAIL_RF_STATE_RX,
+        },
+        .txTransitions = {
+            RAIL_RF_STATE_RX,
+            RAIL_RF_STATE_RX,
+        }},
+    .timings = {
+        .idleToRx = 100,         
+        .idleToRx = 192 - 10,    
+        .idleToTx = 100,         
+        .rxToTx = 192,           
+        .rxSearchTimeout = 0,    
+        .txToRxSearchTimeout = 0,
+    },
+    .framesMask = RAIL_IEEE802154_ACCEPT_STANDARD_FRAMES, 
+    .promiscuousMode = false,                             
+    .isPanCoordinator = false,                            
+};
+
+CORE_irqState_t CORE_EnterCritical(void)
+{
+    LOG_DBG("Enter crit");
+    return irq_lock();
+}
+
+void CORE_ExitCritical(CORE_irqState_t irqState)
+{
+    (void)irqState;
```

Comment:
Why are you doing this? (Which is what would do ARG_UNUSED() macro if I remember well)